TSRM/TSRM.c

```diff
@@ -170,30 +177,33 @@ TSRM_API void tsrm_shutdown(void)
 {
 	int i;
 
-	if (tsrm_tls_table) {
-		for (i=0; i<tsrm_tls_table_size; i++) {
-			tsrm_tls_entry *p = tsrm_tls_table[i], *next_p;
+	if (!in_main_thread) {
+		/* ensure singleton */
+		return;
+	}
```

Comment:
Changes/fix related to "tsrm_tls_table" and "in_main_thread" look unrelated to interned strings. Lets commit this part separately.